src/minters/signature/ZoraCreatorSignatureMinterStrategy.sol

```diff
@@ -0,0 +1,248 @@
+import {IMinter1155} from "../../interfaces/IMinter1155.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleStrategy} from "../SaleStrategy.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleCommandHelper} from "../utils/SaleCommandHelper.sol";
+import {LimitedMintPerAddress} from "../utils/LimitedMintPerAddress.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {IReadableAuthRegistry} from "../../interfaces/IAuthRegistry.sol";
+
+/*
+
+
+             ░░░░░░░░░░░░░░              
+        ░░▒▒░░░░░░░░░░░░░░░░░░░░        
+      ░░▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░      
+    ░░▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░    
+   ░▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░    
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░░  
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░░░  
+  ░▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░░  
+  ░▓▓▓▓▓▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░  
+   ░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░  
+    ░░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░    
+    ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒░░░░░░░░░▒▒▒▒▒░░    
+      ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒░░      
+          ░░▓▓▓▓▓▓▓▓▓▓▓▓▒▒░░░          
+
+               OURS TRULY,
+
+
+    github.com/ourzora/zora-1155-contracts
+
+ */
+
+/// @title ZoraCreatorSignatureMinterStrategy
+/// @notice Mints tokens based on signature created by an authorized signer
+/// @author @oveddan
+contract ZoraCreatorSignatureMinterStrategy is Enjoy, SaleStrategy, LimitedMintPerAddress, EIP712 {
+    using SaleCommandHelper for ICreatorCommands.CommandSet;
+
+    /// @notice General signatue sale settings
+    struct SalesConfig {
+        /// @notice Registry that decides which account is authorized to sign signatures
+        /// to mint for the sale
+        IReadableAuthRegistry authorizedSignatureCreators;
+    }
+
+    // target -> settings
+    mapping(address => SalesConfig) signatureSaleSettings;
+    // target contract -> unique nonce -> if has been minted already
+    mapping(address => mapping(bytes32 => bool)) private minted;
+
+    /// @notice Event for sale configuration updated
+    event SaleSet(address indexed mediaContract, SalesConfig signatureSaleSettings);
+
+    error SaleEnded();
+    error SaleHasNotStarted();
+    error WrongValueSent(uint256 expectedValue, uint256 valueSent);
+    error InvalidSignature();
+    error AlreadyMinted();
+    error MissingFundsRecipient();
+    error Expired(uint256 expiration);
+
+    bytes32 constant REQUEST_MINT_TYPEHASH =
+        keccak256(
+            "requestMint(address target,uint256 tokenId,bytes32 nonce,uint256 quantity,uint256 pricePerToken,uint256 expiration,address mintTo,address fundRecipient)"
+        );
+
+    /// @notice ContractURI for contract information with the strategy
+    function contractURI() external pure override returns (string memory) {
+        return "https://github.com/ourzora/zora-1155-contracts/";
```

Comment:
is this the correct url, since it's a private repo?

Reply:
yes